### ISSUE TITLE
Users info: add a filter by season

### ISSUE THREAD

**Issue by mkalininait** [rails-girls-summer-of-code/rgsoc-teams#695]
`/users/info`

Add the third filter by season: either a checkbox "In current season" or radiobuttons for all ever existed seasons.

Motivation: we want to send an email to all students who applied this year about the crowdfunding campaign.

**Comment by Anagha22** [comment]
@mkalininait Hello, we the team Bundledore would like to work on this issue.

**Comment by mkalininait** [comment]
Hey @Anagha22, sounds like a good idea. Please ping me if you need details on the requirements.

**Comment by Anagha22** [comment]
Hi @mkalininait I have added a filter with a radio button for "current season" and added filters like 
"Applied " and  "Selected".Applied lists out the teams in current season and selected lists out none. Is this the changes that you were expecting? Please give me your suggestions.

**Comment by mkalininait** [comment]
@Anagha22 Could you please upload a screenshot here in case it's possible to make one?

**Comment by Anagha22** [comment]
Hey @mkalininait  I have attached the screenshot here. Please help me with this.
![user_info](https://cloud.githubusercontent.com/assets/9162231/24251033/cde05456-0ffe-11e7-8178-90943e1b5d71.png)

**Comment by mkalininait** [comment]
Hi @Anagha22 
Thank you! It looks good to me. One suggestion: is it possible to make a checkbox instead of the radiobutton for the season filter?

I can't judge if your filters work correctly or not without seeing the code and testing it. 
I'd say that the "Applied" filter should list **all** teams that applied from **all** seasons. If the "Current season" filter is on, then, of course, we should see only teams from the current season. Same applies to the "Selected" filter. 

Looking forward to seeing your PR!

**Comment by Anagha22** [comment]
@mkalininait Yeah, I will work on that and will send the PR as soon as possible :)

**Comment by mkalininait** [comment]
@Anagha22 Thank you, no rush! :)

**Comment by mkalininait** [comment]
Hi @Anagha22 Do you still want to send your PR or should we take this issue over?

**Comment by F3PiX** [comment]
Looking at the goal: 
> we want to send an email to all students who applied this year about the crowdfunding campaign.  

Wouldn't it make more sense to add the filter to the Mailer view (or even the Exporters)?
If that would solve the problem, I'd vote for that. (Because I am not a big fan of stapling filters in the view. I consider adding a third filter a 'design smell', and a sign that we should rethink the filtering options.)
AFAIK you can't DO anything with the filtered students in the `students/info` screen, can you? What do you have to do with the filter results in order to actually emailing them?

**Comment by mkalininait** [comment]
@F3PiX 
>AFAIK you can't DO anything with the filtered students in the students/info screen, can you? What do you have to do with the filter results in order to actually emailing them?

You probably mean `users/info`, right? You can't do anything directly in the Teams App. What we would do is: a) copy email addresses to send emails from HelpScout (small groups, e.g. accepted teams); b) copy email addresses to send emails from MailChimp (large groups, e.g. all students who applied in the current season); c) copy/print addresses to send swag (accepted teams); d) copy a view and paste into a google spreadsheet for any other purposes.

> Wouldn't it make more sense to add the filter to the Mailer view (or even the Exporters)?

We can make an export (or multiple exports) for this, of course. But then the `users/info` page will stay useless.
What is the Mailer view?

**Comment by mkalininait** [comment]
Oh, btw, I've just discovered another page `teams/info`, and it looks pretty useless, too.

**Comment by F3PiX** [comment]
Yes, users/info. 
> What we would do is: a) copy email addresses...
How do you copy them now? One by one?

**Comment by mkalininait** [comment]
@F3PiX 
There are a few use cases:
1) We need contacts of all students who applied in the current season. What I did this year:
 - I got the export Teams#current; it gives you contacts of all teams ever existed;
 - I got the export Applications#current; it gives you all applications of the current season, but without contacts;
 - I wrote a simple script, that filters the list of teams based on the team names from the list of applications.
2) We need contacts of students who have been accepted:
 - Theoretically you can use the export Users#current_students, but we usually need their contacts **before** they are marked as accepted in the teams app;
 - Thus we take them one by one from the user profiles;
 - Having them in the applications export ([here](https://teams.railsgirlssummerofcode.org/rating/applications)) would help.
3) Then we also need contacts of rejected teams, but it's relatively easy to make from the list of all students who applied minus all students who were accepted. Although it's still a manual work.

All contacts are then stored in Google Spreadsheets.

I may be missing something (want to sleeeep), but hopefully nothing important.

It may look like the time we spent on fetching the data this year is comparable to the time we could spend writing a new data export or something :D

**Comment by F3PiX** [comment]
a) So, you don't use the mailing functionality? (.../mailings/new). Is anybody using that at all? 
 
b) Re use case 1) You happy when we start with adding a filter with the Applications#current including the email addresses? (Or what do you need as 'contacts'?)

c) Re: use case 2) and 3) : Are they relevant _before_ the start of the current season?

**Comment by mkalininait** [comment]
@F3PiX 

a) OMG, I didn't even know about this feature 🙈  It looks useful, but I never heard of anyone in the team using it in 2016-2017. What about you @alicetragedy @anikalindtner ?

b) I think that having names and emails of students and coaches in the export of current applications would help a lot. It will actually solve the use cases 2) and 3) as well.

**NOTE 1:** We actually use this export of applications -> https://teams.railsgirlssummerofcode.org/rating/applications the "Download as CSV" button at the bottom of the page. I don't know if this is a part of what you call the Exporter or not.

**NOTE 2:** Not entirely related, but I want to mention it here, too. There is a major flaw in how we save data about an application when it is submitted. We have data from two students, but we don't save their names (or ids), thus we don't know who is Student 1 and who is Student 2 in an application. Thus, until we fix this bug, we can show student names and emails in the table/export of applications, but we won't know, who is Student 1 and who is Student 2.

c) Yep, they are. We have to contact the students whom we would like to select *before* they are selected. But the solution mentioned in b) will help us to do it.

**Comment by F3PiX** [comment]
@mkalininait 
:-) There are many jewels hidden in the teams app.
I am looking for the Smallest Change with Biggest Impact. Because the season starts shortly. So let's take it step by step. 

Two more questions:
1.
As I understand it now, if you can download an csv with 
- [ ] names and email addresses of the coaches
- [ ] names and email addresses of  all the students who applied: excepted, volunteers and not excepted
It would give you a basic 'contacts' spreadsheet, and you can import that in the different  Spreadsheets and apps. 
Right?

2.
Can you please test the output of; `/orga/exports`, -> `Exporters::Teams#current` 
You said before: 
 > I got the export Teams#current; it gives you contacts of _all teams ever existed_;
That is not what the code _intends_ to do.
It intends to give the teams in the 2017 season with names and email addresses of all the members.
Can you check this in the output please? 
(3)
I also would like to know if it returns the accepted teams only, or the not accepted teams as well.

I think we are already close to a solution. I wouldn't be surprised if we just need some finetuning in the existing filters. 🙌 
In next iterations we can work on the overall workflow.

**Comment by alicetragedy** [comment]
> OMG, I didn't even know about this feature 🙈 It looks useful, but I never heard of anyone in the team using it in 2016-2017. What about you @alicetragedy @anikalindtner ?

I used it a few times this year already (to ask past participants what they are up to and to send reminders about the project submission deadlines to past mentors); we also used it a lot in 2016: reminders regarding all the feedback forms, an info email about the video, and info about the RGSoC celebration events around the world. 

So, to make a long story short I think it's a very useful feature during the summer, or to communicate specifically with a certain group of people (i.e. sending feedback forms)

**Comment by F3PiX** [comment]
@alicetragedy Cool! And it is only orga's who should use the mailer, right? Or do you want to have it available for supervisors, for instance? And maybe other non-orga users?
If it is only for orga's, it would make sense to move it to the orga namespace.

**Comment by alicetragedy** [comment]
@F3PiX I don't think it makes sense for anyone other than orgas to use it in its current state (the email comes from our orga email address, which is where all replies to the email will go). 
In order for it to be useful for supervisors, for example to get in touch with their own team, quite a lot has to happen on the filter side of things — and all mails sent are visible to everyone, which makes “private” communication difficult. Fixing all of that (to basically “reinvent email”) would I think be too much hassle, for a use case we don't really have. I hope this answers your question properly!

So yeah, I agree that it would make sense to move it to the orga namespace.

**Comment by mkalininait** [comment]
@F3PiX 

2\) Regarding `Exporters::Teams#current`. I've tested it again. You are right, I was mistaken that it gives you all teams ever existed, BUT it looks like it gives you all teams of the current season **regardless** if they applied or not. At least I think this is how it works. I see 294 teams in the exported list, while we got only ~189 applications this year.

1\) Almost. Would it be one csv? 
> names and email addresses of the coaches

Of the accepted teams (sponsored and volunteer).

> names and email addresses of all the students who applied: excepted, volunteers and not excepted

Do you mean *accepted* instead of *excepted*? We need all the students who: a) applied, b) have been accepted (sponsored, volunteer separately), c) have been rejected (= not accepted).

One comment: if this data comes separately from the application data, we will have to make a list of contacts of accepted teams manually. Use case: we export the Applications Table, we import it in Google Docs, we run a complicated selection process, select top-20 teams and then run interviews with them. To run these interviews we need their contacts. We mark these teams as Accepted after the interviews, if everything goes well.

**Comment by F3PiX** [comment]
LOL yes, accepted, not excepted, sorry.
Thanks for testing!! 
It makes sense to have an exporter for the current_teams who actually applied. And also add  their 'status'.
But I am getting confused. Isn't the application phase over? I so understand that the Exporters are not fully optimized yet for all the use cases, but let's take it step by step. I am looking to make all of this actionable. 
I _expected_ that a 'general contacts sheet' for the _accepted_ teams would be the most helpful overall, but it is not?  
If not, could you pinpoint one use case that will help you the most in the weeks upto the start of the season?  And then we can take it from there.

**Comment by mkalininait** [comment]
@F3PiX 
> It makes sense to have an exporter for the current_teams who actually applied. And also add their 'status'.

+1

> Isn't the application phase over?

Oh, it looks like I confused you. The application and selection phases are over indeed. We don't need any exports with email addresses this year anymore, because we already have them. But we will **definitely** need them next year. What we still need this year is the list of *postal* addresses of the accepted teams. I think the best would be having the list of all team members (incl. coaches, mentors and supervisors), but with their role in the table (so that we could filter it by role). 
Maybe we could make a brand new export called contacts#current where we list team members of all teams which applied this year with their team name, status, role, email and postal address? WDYT @alicetragedy ?

> I expected that a 'general contacts sheet' for the accepted teams would be the most helpful overall, but it is not?

Yes, probably. It doesn't cover only one use case (as far as I can tell): when we need contacts of students whom we want to interview during the selection phase (what I mentioned in my previous comment). We don't have this issue this year anymore, but we will have it next year.

Does it make sense?

**Comment by F3PiX** [comment]
Well, let's focus on this year first. (I want to start working on the conferences in time for this season, so I don't want to take on things for next year right now.) If @alicetragedy agrees, we'll go for the "swag exporter" first.
